frontend/src/pages/CartPage.js

```diff
@@ -0,0 +1,306 @@
+import React, { Component } from 'react'
+import { Link } from 'react-router-dom'
+
+import withLayout from './withLayout'
+import { Popover } from '../components/common'
+import { getColumnName } from '../utils/helpers'
+
+class Cart extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      showChooseCols: false,
+      isAverage: localStorage.getItem('meta-column-type') !== 'recent',
+      courses: [],
+      excludedCourses: [],
+      boxValues: ['N/A', 'N/A', 'N/A', 'N/A'],
+      boxLabels: [
+        'rCourseQuality',
+        'rInstructorQuality',
+        'rDifficulty',
+        'rWorkRequired',
+      ],
+    }
+
+    // TODO: Move regeneration logic into Redux or React Context
+    this.regenerateRatings = this.regenerateRatings.bind(this)
+  }
+
+  componentDidMount() {
+    window.addEventListener('storage', this.regenerateRatings)
+    this.regenerateRatings()
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('storage', this.regenerateRatings)
+  }
+
+  regenerateRatings() {
+    const courses = Object.keys(localStorage)
+      .filter(k => !k.startsWith('meta-'))
+      .map(k => {
+        const out = JSON.parse(localStorage.getItem(k))
+        if (typeof out !== 'object') {
+          return null
+        }
+        const typeDict = {}
+        if (typeof out.info !== 'undefined') {
+          out.info.forEach(v => {
+            typeDict[v.category] = v
+          })
+          out.info = typeDict
+        }
+        out.course = k
+        return out
+      })
+      .filter(a => a !== null)
```

Comment:
This could be a helper function

Reply:
I think I'll leave this to the ticket involving moving localStorage into a store file/potentially looking into Redux?

Reply:
nvm, moved the func